feat: moved types from sdk project into runtime

colincasey opened the pull request:
Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `@heroku/sf-fx-sdk-nodejs`.

[GUS-W-11944771](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000019Rqb8YAC/view)

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `@heroku/sf-fx-sdk-nodejs`.

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `@heroku/sf-fx-sdk-nodejs`.

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `@heroku/sf-fx-sdk-nodejs`.

joshwlewis commented on src/types.ts:
```diff
@@ -0,0 +1,309 @@
+/*
```
For folks using typescript, wouldn't they need to import these types with a suffix? Like this:

```typescript
import { InvocationEvent, Context, Logger, RecordQueryResult } from 'sf-fx-runtime-nodejs/types';
```

Would it make more sense to name this file `index.ts` instead, so they could be imported from `'sf-fx-runtime-nodejs'` (without a suffix)?

joshwlewis replied:
Or if we don't move the file, we could at least re-export from `index.ts`?

joshwlewis reviewed (approved):
Prob need root-level exports for the types, but I won't hold you up for re-review on that. Looks good otherwise!

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `@heroku/sf-fx-sdk-nodejs`.

[GUS-W-11944771](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000019Rqb8YAC/view)

colincasey committed:
Merge branch 'main' into feat/W-11944771_include_sdk_types

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `sf-fx-sdk-nodejs`.

[GUS-W-11944771](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000019Rqb8YAC/view)

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `sf-fx-sdk-nodejs`.

[GUS-W-11944771](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000019Rqb8YAC/view)

colincasey committed:
feat: moved types from sdk project into runtime

Since we are now requiring the runtime as an explicit dependency for Node.js Function projects then we no longer need the overhead of maintaining separate SDK type definitions in `sf-fx-sdk-nodejs`.

[GUS-W-11944771](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000019Rqb8YAC/view)
